index.bs

```diff
@@ -8923,6 +8930,79 @@ The [=remote end steps=] given |session|, and |command parameters| are:
 
 </div>
 
+#### The input.setFiles Command ####  {#command-input-setFiles}
+
+The <dfn export for=commands>input.setFiles</dfn> command sets a given file
+input with a given set of file paths.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      input.SetFiles = (
+        method: "input.setFiles",
+        params: input.SetFilesParameters
+      )
+
+      input.SetFilesParameters = {
+        context: browsingContext.BrowsingContext,
+        element: script.SharedReference,
+        files: [*text]
+      }
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl">
+     EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for input.setFiles">
+
+The [=remote end steps=] given |session|, and |command parameters| are:
+
+1. Let |context id| be the value of the <code>context</code> field of
+   |command parameters|.
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+   with |context id|.
+
+1. Let |top-level context| be |context|'s [=top-level browsing context=].
+
+<!-- converting to realm here is kind of silly since we immediately convert back
+      in [=deserialize remote reference=] -->
+1. Let |document| be |top-level context|'s [=active document=].
+
+1. Let |environment settings| be the [=environment settings object=] whose
+   [=relevant global object=]'s <a>associated <code>Document</code></a> is
+   |document|.
+
+1. Let |realm| be |environment settings|' [=realm execution context=]'s
+   Realm component.
+
+1. Let |reference| be the value of the <code>element</code> field of
+   |command parameters|.
+
+1. Let |element| be the result of [=trying=] to [=deserialize remote reference=]
+   with |reference|, |realm|, and |session|.
+
+1. If |element| doesn't implement {{HTMLInputElement}} return [=error=] with [=error code=]
+   [=unable to set file input=].
+
+1. If the [=input type=] of |element| doesn't match <code>"file"</code> return [=error=] with [=error code=]
+   [=unable to set file input=].
+
+1. Let |files| be the value of the <code>files</code> field of |command parameters|.
+
+1. [=Update the file selection=] for |element| with |files| as if a user
```

Comment:
`as if a user selected |files| through the [=show the picker|file picker=] prompt` seems redundant? as it's not clear if it's actionable and in what way?

Reply:
I think the problem here is this is implementation-specific, so I'm not sure if we can be more specific. @whimboo Is this sufficient? From Chromium's standpoint, this has a precise meaning because we are able to set an input with an arbitrary set of paths provided by either CDP or the file picker prompt.

Reply:
Note: `Update the file selection` points to a spec-defined action.

Reply:
Please also see https://github.com/w3c/webdriver/issues/1742 for an open issue with classic. We should make sure to implement it correctly in BiDi.